tests/admintools.test.js

```diff
@@ -0,0 +1,212 @@
+
+var nowData = {
+  sgvs: [
+    { mgdl: 100, mills: Date.now(), direction: 'Flat', type: 'sgv' }
+  ]
+};
+
+var someData = {
+  '/api/v1/devicestatus.json?count=500': [
+    {
+    '_id': {
+        '$oid': '56096da3c5d0fef41b212362'
+    },
+    'uploaderBattery': 37,
+    'created_at': '2015-09-28T16:41:07.144Z'
+    },
+    {
+    '_id': {
+        '$oid': '56096da3c5d0fef41b212363'
+    },
+    'uploaderBattery': 38,
+    'created_at': '2025-09-28T16:41:07.144Z'
+    }
+  ],
+  '/api/v1/treatments.json?&find[created_at][$gte]=': [
+      {
+        '_id':  '5609a9203c8104a8195b1c1e',
+        'enteredBy': '',
+        'eventType': 'Carb Correction',
+        'carbs': 3,
+        'created_at': '2025-09-28T20:54:00.000Z'
+      }
+    ],
+  '/api/v1/entries.json?&find[date][$gte]=': [
+      {
+        '_id': '560983f326c5a592d9b9ae0c',
+        'device': 'dexcom',
+        'date': 1543464149000,
+        'sgv': 83,
+        'direction': 'Flat',
+        'type': 'sgv',
+        'filtered': 107632,
+        'unfiltered': 106256,
+        'rssi': 178,
+        'noise': 1
+      }
+    ]
+  };
+
+
+describe('admintools', function ( ) {
+  var self = this;
+
+  before(function (done) {
+    benv.setup(function() {
+      self.$ = require('jquery');
+      self.$.localStorage = require('./fixtures/localstorage');
+
+      self.$.fn.tipsy = function mockTipsy ( ) { };
+
+      self.$.fn.dialog = function mockDialog (opts) {
+        function maybeCall (name, obj) {
+          if (obj[name] && obj[name].call) {
+            obj[name]();
+          }
+
+        }
+        maybeCall('open', opts);
+
+        _.forEach(opts.buttons, function (button) {
+          maybeCall('click', button);
+        });
+      };
+
+      var indexHtml = read(__dirname + '/../static/admin/index.html', 'utf8');
+      self.$('body').html(indexHtml);
+
+      //var filesys = require('fs');
+      //var logfile = filesys.createWriteStream('out.txt', { flags: 'a'} )
+      
+      self.$.ajax = function mockAjax (url, opts) {
+        //logfile.write(url+'\n');
+        //console.log(url,opts);
+        if (opts && opts.success && opts.success.call) {
+          if (url.indexOf('/api/v1/treatments.json?&find[created_at][$gte]=')===0) {
+            url = '/api/v1/treatments.json?&find[created_at][$gte]=';
+          }
+          if (url.indexOf('/api/v1/entries.json?&find[date][$gte]=')===0) {
+            url = '/api/v1/entries.json?&find[date][$gte]=';
+          }
+          return {
+            done: function mockDone (fn) {
+                if (someData[url]) {
+                  console.log('+++++Data for ' + url + ' sent');
+                  opts.success(someData[url]);
+                } else {
+                  console.log('-----Data for ' + url + ' missing');
+                  opts.success([]);
+                }
+              fn();
+              return self.$.ajax();
+            },
+            fail: function mockFail (fn) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: ['fn' is defined but never used.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1039407048/issues/source?bid=2452379&fileBranchId=2878156#l110)